pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -207,16 +217,41 @@ func (p *Pod) Stop(ctx context.Context, c client.Client, _ []jobframework.PodSet
 }
 
 func SetupIndexes(ctx context.Context, indexer client.FieldIndexer) error {
+	if err := indexer.IndexField(ctx, &kueue.Workload{}, WorkloadNameKey, func(o client.Object) []string {
+		wl, ok := o.(*kueue.Workload)
+		if !ok {
+			return nil
+		}
+		return []string{wl.ObjectMeta.Name}
+	}); err != nil {
+		return err
+	}
+
 	return jobframework.SetupWorkloadOwnerIndex(ctx, indexer, gvk)
 }
 
 func (p *Pod) Finalize(ctx context.Context, c client.Client) error {
```

Comment:
Finalizers are tricky.

What sometimes happens is that you can enter this "Finalize" function, but not all the pods are there, and we are left with a Pod with a stuck finalizer.

But if we see this happens in integration tests, we can fix in a follow up.